ethereum/contracts/Splitter.sol

```diff
@@ -1,28 +1,43 @@
-pragma solidity >= 0.5.7;
-pragma experimental ABIEncoderV2;
+pragma solidity >=0.5.8;
 
 contract Splitter {
-  mapping(address => bool) accounts;
+  mapping(address => bool) blackAccounts;
   address owner;
 
   constructor() public {
     owner = msg.sender;
   }
 
-  event Transfer(address from, bool status, uint balance1, uint balance2);
+  event Transfer(address from, address receiver1, address receiver2);
 
-  function sendEther(address[] memory receivers) public payable {
+  function sendEther(address payable receiver1, address payable receiver2) public payable {
     // One transfer call needs 2300 gas
     require(msg.sender.balance + 4600 > msg.value, 'A sender should have enough balance');
-    require(receivers.length == 2, 'The number of receivers should be 2');
-    require(msg.value % 2 == 0, 'The value should be evenly divisible');
     require(msg.value > 0, 'A given value should be bigger than 0');
-    require(msg.sender != receivers[0] && msg.sender != receivers[1], 'A sender should not be one of receivers');
+    require(msg.sender != receiver1 && msg.sender != receiver2, 'A sender should not be one of receivers');
+
+    if (blackAccounts[receiver1] || blackAccounts[receiver2]) {
+      revert('A receiver is in black list');
+    }
 
     uint value = msg.value / 2;
-    address(uint160(receivers[0])).transfer(value);
-    address(uint160(receivers[1])).transfer(value);
 
-    emit Transfer(msg.sender, true, receivers[0].balance, receivers[1].balance);
+    // TODO: security/no-send: Consider using 'transfer' in place of 'send'.
```

Comment:
It is almost always better to use `.transfer`- "Fail early. Fail hard." For now, I think we should stick with this because I might be able to lead you to something I hope you will discover on your own. 

You are using `.send` and dealing with the failure case. 

You cannot revert. (See other comment). 

Suppose you want to deal with the failure case. You _must_ provide another way for the receiver to obtain their funds because:

1. the contract has accepted the funds
2. if there is no way to retrieve those funds, then they are effectively destroyed. 

A solution to that problem is do some sort of accounting and let the receiver ask for their money, i.e. `function withdraw()`

Play with that and let's see where that takes us.

Reply:
I see what you mean, like a bank(contract), the bank will take money from a sender and every receiver can withdraw at any time.